samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
```

Comment:
Discussed offline. For remote execution plan var is defined with the folder ./plan, so it doesn't expose to the user. For local runner, it requires setting up this var as if the user want to write the plan to a file. We will see how this works out and there will be future improvement for the plan folder.